internal/operators/ocs/validations.go

```diff
@@ -69,24 +71,25 @@ func setOperatorStatus(cluster *models.Cluster, status string) error {
 func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
 	var status string
 	hosts := cluster.Hosts
-
+	var err error
 	if int64(len(hosts)) < o.OCSRequiredHosts {
 		status = "Insufficient hosts to deploy OCS. A minimum of 3 hosts is required to deploy OCS. "
 		err := setOperatorStatus(cluster, status)
 		if err != nil {
-			o.log.Fatal("Failed to set Operator status ", err)
+			o.log.Error("Failed to set Operator status ", err)
 			return false
 		}
 		o.log.Info("Setting Operator Status ", status)
 		return false
 	}
-	var CPUCount int64 = 0                       //count the total CPUs on the cluster
-	var TotalRAM int64 = 0                       // count the total available RAM on the cluster
-	var DiskCount int64 = 0                      // count the total disks on all the hosts
-	var HostsWithDisks int64 = 0                 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks                           // to claim for the PV created by LocalVolumeSet
+	var cpuCount int64 = 0       //count the total CPUs on the cluster
+	var totalRAM int64 = 0       // count the total available RAM on the cluster
+	var diskCount int64 = 0      // count the total disks on all the hosts
+	var hostsWithDisks int64 = 0 // to determine total number of hosts with disks. OCS requires atleast 3 hosts with non-bootable disks
+	var insufficientHosts []string
 	if int64(len(hosts)) == o.OCSRequiredHosts { // for only 3 hosts, we need to install OCS in compact mode
 		for _, host := range hosts {
-			err := o.nodeResources(host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
+			err, insufficientHosts = o.nodeResources(cluster, host, &cpuCount, &totalRAM, &diskCount, &hostsWithDisks, insufficientHosts)
```

Comment:
Does this need to be returned if it is already being passed to the function as an argument?

Reply:
fixed